Objects/floatobject.c

```diff
@@ -2192,6 +2192,18 @@ PyFloat_Pack4(double x, char *data, int le)
         if (isinf(y) && !isinf(x))
             goto Overflow;
 
+        /* correct y if x was a sNaN, transformed to qNaN by conversion */
+        if (isnan(x)) {
+            uint64_t v;
+
+            memcpy(&v, &x, 8);
+            if ((v & (1ULL << 51)) == 0) {
+                uint32_t *py = (uint32_t *)&y;
```

Comment:
I'm not sure that this code works well with strict aliasing. You might need to use an union or memcpy() to be 100% safe.